lib/OpenQA/Client/Archive.pm

```diff
@@ -0,0 +1,219 @@
+# Copyright (C) 2018 SUSE Linux Products GmbH
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package OpenQA::Client::Archive;
+use Mojo::Base 'OpenQA::Client::Handler';
+
+use Mojo::Exception;
+use Mojo::File 'path';
+use Mojo::URL;
+use Carp 'croak';
+use JSON;
+
+use Data::Dump;
+
+my $job;
+my $path;
+my $url;
+my $options;
+
+sub run {
+
+    #We'he only taking one argument
+    my ($self, $options) = @_;
+
+    croak 'Options must be a HASH ref' unless ref $options eq 'HASH';
+    croak 'Need a file to upload in the options!' unless $options->{url};
+
+    $url = Mojo::URL->new($options->{url});
+    my $req = $self->client->get($url);
+    my $res = $req->res;
+
+    if ($res->code eq 200) {
+        $job = $res->json->{job} || die("No job could be retrieved");
+        #We have a job now, make a directory in $pwd by default
+        $path = path($options->{archive})->make_path;
+        #we can't backup repos, so don't even try
+        delete($job->{assets}->{repos});
+        $path->path('testresults/thumbnails')->make_path if $options->{'with-thumbnails'};
+        $self->download_assets unless $options->{'skip-download'};
```

Comment:
@coolo regarding your previous comment about the opt-in for download of assets:

I tried by using max_size_message_size but for some reason didn't work in my first tests... Will do a quick shot to see if there's something else messing around here or it's just my code not doing something right.

Reply:
@coolo We're good to go here :)